packages/utils/upgrade/resources/codemods/5.0.0/strapi-public-interface.code.ts

```diff
@@ -0,0 +1,126 @@
+import { Transform, JSCodeshift, Collection } from 'jscodeshift';
+
+/*
+This codemod transforms @strapi/strapi imports to use the new public interface.
+
+ESM
+Before:
+
+import strapi from '@strapi/strapi';
+strapi();
+
+After:
+
+import { createStrapi } from '@strapi/strapi'; // keeps the default import
+createStrapi();
+
+---
+
+Common JS
+Before:
+
+const strapi = require('@strapi/strapi');
+strapi();
+
+After:
+
+const strapi = require('@strapi/strapi');
+strapi.createStrapi();
+
+*/
+
+const transformStrapiImport = (root: Collection, j: JSCodeshift) => {
+  root.find(j.ImportDefaultSpecifier).forEach((path) => {
+    if (path.parent.value.source.value === '@strapi/strapi') {
```

Comment:
Can't this be part of the `find`'s `filter` clause?

Reply:
afaik, you can't filter on the parent in the find

Reply:
Missed the parent part, my bad